.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,171 @@
+          echo "Startup complete"
+      - name: install and start containerd
+        shell: bash
+        run: sudo scripts/install-containerd
+      - name: start ecs local endpoint
+        shell: bash
+        run: scripts/start-ecs-local-endpoint
+      - name: ip forwarding # awslogs driver hardcodes "169.254.170.2" as the aws credential endpoint ip so need to forward to local endpoint
+        shell: bash
+        run: sudo scripts/ip-forwarding
+      - name: build
+        run: sudo make build
+      - name: Run benchmark
+        run: sudo -E make test-benchmark-for-awslogs | tee benchmark-for-awslogs.txt
+        shell: bash
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: Benchmark for awslogs
+          tool: 'go'
+          benchmark-data-dir-path: "dev/bench/awslogs/ubuntu"
+          output-file-path: benchmark-for-awslogs.txt
+      - name: Push benchmark result
+        run: git push 'https://github.com/aws/shim-loggers-for-containerd.git' gh-pages:gh-pages
+  benchmark-for-fluentd:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ] # TODO: Add Windows e2e tests: https://github.com/aws/shim-loggers-for-containerd/issues/68
+    name: Benchmark / fluentd / ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
+          cache: false
+      - name: install and start containerd
+        shell: bash
+        run: sudo scripts/install-containerd
+      - name: start fluentd local endpoint
+        shell: bash
+        run: |
+          : # not using github action env because env map cannot be defined in terms of other vars in the map. public.ecr.aws/docker/library/fluentd:v1.16-debian-1
+          FLUENTD_LOG_DIR=${GITHUB_WORKSPACE}/fluentd-logs
+          FLUENTD_PORT=24224
+          FLUENTD_IMAGE=public.ecr.aws/docker/library/fluentd:v1.16-debian-1
+          : # ECR Public anonymous pull rate quota is 1 per sec. https://docs.aws.amazon.com/AmazonECR/latest/public/public-service-quotas.html.
+          : # Running multiple jobs at the same time may exceed rate limit. Retry 3 times to prevent flaky behavior.
+          count=0
+          until [ $count -ge 3 ]
+          do
+            docker pull $FLUENTD_IMAGE && break
+            count=$((count+1))
+            [ $count -eq 3 ] && echo "Pull command failed after 3 attempts" && exit 1
+            sleep 3
+          done
+          : # Fluentd container is not using root user so need 777 to make it writable. https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#docker-container-filesystem
+          sudo mkdir -m 777 $FLUENTD_LOG_DIR
+          docker run -d -p $FLUENTD_PORT:24224 -p $FLUENTD_PORT:24224/udp -v $FLUENTD_LOG_DIR:/fluentd/log $FLUENTD_IMAGE
```

Comment:
Same here, probably just wrap this in a script for CI and Benchmark workflow to share.